### PR TITLE
Hw 2

### PR DESCRIPTION
**Pull request by safonovaA**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -1,7 +1,24 @@
 import { User, Product } from './models';
 import config from './config/config';
+import { DirWatcher } from './dirwatcher';
+import path from 'path';
```
Comment (ubcent) [inline src/app.js]: I'd recommend you should separate imports of built-in, third-party and your own modules

File: src/dirwatcher.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        .then((files) => {
+          this.prevFiles = files;
+          this.isFirstCheck = false;
+        }) : 
+      this.compareFiles(path)
+        .then((files) => {
+          this.prevFiles = files;
+        });
+    setTimeout(() => {
+      console.log('check completed, watch')
```
Comment (ubcent) [inline src/dirwatcher.js]: Could you please remove unnecessary console.logs

File: src/importer.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  import(path) {
+    console.log('async');
+    return new Promise((resolve, reject) => {
+      readdir(path)
+        .then((files) => {
+          resolve(Promise.all(files.map((file) => {
+            return new Promise((resolve) => {
+              readFile(`${path}/${file}`)
+                .then((data) => {
+                  const buf = Buffer.from(data);
```
Comment (ubcent) [inline src/importer.js]: Looks strange :)

File: src/importer.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        })
+        .catch((err) => reject(err))
+    })
+  }
+
+  importSync(path) {
+    const files = fs.readdirSync(path);
+    return files.map((file) => {
+      const data = fs.readFileSync(`${path}/${file}`);
+      const buf = Buffer.from(data);
```
Comment (ubcent) [inline src/importer.js]: See above

**Review by ubcent — APPROVED** [review]

File: src/importer.js (hunk 3 above)
Comment (safonovaA) [reply src/importer.js]: Fixed:)

File: src/importer.js (hunk 4 above)
Comment (safonovaA) [reply src/importer.js]: Fixed

File: src/dirwatcher.js (hunk 2 above)
Comment (safonovaA) [reply src/dirwatcher.js]: Done

File: src/app.js (hunk 1 above)
Comment (safonovaA) [reply src/app.js]: Done

**Pull request merged**